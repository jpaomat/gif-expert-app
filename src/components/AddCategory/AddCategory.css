input {
    color: grey;
    font-size: 1.3rem;
    width: 100%;
}