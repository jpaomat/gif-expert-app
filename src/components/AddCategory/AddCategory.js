import React, { useState } from 'react';
import PropTypes from 'prop-types';

import './AddCategory.css';


export const AddCategory = ({setCategories}) => {
    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => setInputValue(e.target.value);
    const handleSubmit = (e) => {
        e.preventDefault(); // se usa para que no se recargue la pagina al ejecutar el submit
        if (inputValue.trim().length > 2) {
            setCategories(ctgs=>[inputValue, ...ctgs]);
            setInputValue('');
        }
    }

    return (
        <>
            <form onSubmit={handleSubmit}>
                <input
                    type='text'
                    value={inputValue}
                    onChange={handleInputChange}
                ></input>
                <p>{inputValue}</p>
            </form>
        </>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
