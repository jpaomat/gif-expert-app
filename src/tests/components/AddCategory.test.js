import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory/AddCategory";
import '@testing-library/jest-dom';

describe('Test to AddCategory component', () => {
    const props = {
        setCategories: jest.fn()
    };
    let event;
    let wrapper = shallow(<AddCategory {...props} />); // para que se muestre la ayuda
    beforeEach(() => {
        wrapper = shallow(<AddCategory {...props} />);
        event = {
            target: {
                value: 'toy story'
            },
            preventDefault: jest.fn()
        }
    })

    test('should create component', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should simulate that input value is changing', () => {
        const input = wrapper.find('input');
        input.simulate('change', event);
        expect(wrapper.find('p').text()).toBe(event.target.value);
    })

    test('should that when handleSubmit function is called the setCategories hook not have been called', () => {
        wrapper.find('form').simulate('submit', event);
        expect(props.setCategories).not.toHaveBeenCalled();
    })

    test('should call setCategories hook and clean input value', () => {
        const input = wrapper.find('input');
        input.simulate('change', event);
        wrapper.find('form').simulate('submit', event);
        expect(input.prop('value')).toBe('');
        expect(props.setCategories).toHaveBeenCalledTimes(1);
        expect(props.setCategories).toHaveBeenCalledWith(expect.any(Function));
    })

})
