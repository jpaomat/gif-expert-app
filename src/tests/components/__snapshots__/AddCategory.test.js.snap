// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to AddCategory component should create component 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <p />
  </form>
</Fragment>
`;
